scalding-core/src/main/scala/com/twitter/scalding/Stats.scala

```diff
@@ -122,13 +122,21 @@ object RuntimeStats extends java.io.Serializable {
     }
   }
 
+  private[this] var prevFP: FlowProcess[_] = null
   def addFlowProcess(fp: FlowProcess[_]) {
-    val uniqueJobIdObj = fp.getProperty(UniqueID.UNIQUE_JOB_ID)
-    if (uniqueJobIdObj != null) {
-      uniqueJobIdObj.asInstanceOf[String].split(",").foreach { uniqueId =>
-        logger.debug("Adding flow process id: " + uniqueId)
-        flowMappingStore.put(uniqueId, new WeakReference(fp))
+    if (prevFP != fp) {
```

Comment:
actually, we want:

```scala
if (!(prevFP eq fp)) {
```
because we just want to use reference equality on this check (to make sure we never trigger some expensive structural equals method that could exist or be added to FlowProcess). Sorry, just noticed this.